### DISCUSSION TITLE
compilation error on Suse Tumbleweed aarch64 in saverestore.cpp

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1344] [General]
Installed a fresh Suse Tumbleweed:

Linux localhost.localdomain 5.18.11-1-default #1 SMP PREEMPT_DYNAMIC Fri Jul 15 05:36:11 UTC 2022 (4fcb983) aarch64 aarch64 aarch64 GNU/Linux

NAME="openSUSE Tumbleweed"
# VERSION="20220719"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20220719"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20220719"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"

When configuring with 
 cmake .. -DHDF=OFF -DGRIB=OFF -DQHULL=OFF
and compiling with make I get following error:
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: CMakeFiles/gdl.dir/default_io.cpp.o: in function `Data_<SpDByte>::Read(std::istream&, bool, bool, __rpc_xdr*)':
default_io.cpp:(.text+0x12c): undefined reference to `xdrmem_create'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: default_io.cpp:(.text+0x13c): undefined reference to `xdr_short'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: CMakeFiles/gdl.dir/default_io.cpp.o: in function `Data_<SpDByte>::Write(std::ostream&, bool, bool, __rpc_xdr*)':
default_io.cpp:(.text+0x4b8): undefined reference to `xdrmem_create'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: default_io.cpp:(.text+0x4c8): undefined reference to `xdr_short'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: CMakeFiles/gdl.dir/default_io.cpp.o: in function `Data_<SpDString>::Write(std::ostream&, bool, bool, __rpc_xdr*)':
default_io.cpp:(.text+0xa0c): undefined reference to `xdrmem_create'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: default_io.cpp:(.text+0xa3c): undefined reference to `xdr_short'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: default_io.cpp:(.text+0xb2c): undefined reference to `xdr_short'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: default_io.cpp:(.text+0xb58): undefined reference to `xdr_string'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: CMakeFiles/gdl.dir/default_io.cpp.o: in function `xdr_complex(__rpc_xdr*, std::complex<float>*)':
default_io.cpp:(.text+0x4ba4): undefined reference to `xdr_float'
...
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: saverestore.cpp:(.text+0xc8c0): undefined reference to `xdr_int32_t'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: saverestore.cpp:(.text+0xc914): undefined reference to `xdr_int32_t'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: saverestore.cpp:(.text+0xc928): undefined reference to `xdr_int32_t'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: saverestore.cpp:(.text+0xc958): undefined reference to `xdr_int32_t'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: CMakeFiles/gdl.dir/saverestore.cpp.o: in function `lib::updateNewRecordHeader(__rpc_xdr*, unsigned int)':
saverestore.cpp:(.text._ZN3lib21updateNewRecordHeaderEP9__rpc_xdrj[_ZN3lib21updateNewRecordHeaderEP9__rpc_xdrj]+0x54): undefined reference to `xdr_uint32_t'
/usr/lib64/gcc/aarch64-suse-linux/12/../../../../aarch64-suse-linux/bin/ld: saverestore.cpp:(.text._ZN3lib21updateNewRecordHeaderEP9__rpc_xdrj[_ZN3lib21updateNewRecordHeaderEP9__rpc_xdrj]+0x128): undefined reference to `xdr_opaque'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/gdl.dir/build.make:2604: src/gdl] Error 1
make[1]: *** [CMakeFiles/Makefile2:187: src/CMakeFiles/gdl.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

any guess?

**Comment by alaingdl** [comment]
HI. Did you try with the script in ordre to have all the dependacies ?
If I remember well, xdr asks for a specific packet you may not have
and may be not well detected by CMake ...

**Comment by alaingdl** [comment]
https://github.com/gnudatalanguage/gdl/blob/master/scripts/build_gdl.sh

using the "prep" option to add the packages

then run again your cmake way

A.

**Comment by brandy125** [comment]
Sorry for my ignorance!

Works perfect in this way!

Thanks!
